tests/Moq.Tests/NestedTypeMatchersFixture.cs

```diff
@@ -57,5 +57,23 @@ public void SubstituteTypeMatchers_may_substitute_only_some_parts(Type type, Typ
 		}
 
 		#endregion
+
+		#region Test cases
+
+		[Fact]
+		public void It_IsAnyType_used_as_generic_type_argument_of_method()
+		{
+			var mock = new Mock<IX>();
+			mock.Setup(m => m.Method<It.IsAnyType>(It.IsAny<IEnumerable<It.IsAnyType>>())).Verifiable();
+			mock.Object.Method(new string[] { "a", "b" });
```

Comment:
I'm not actually sure I understand why this should work, given the previous tests for `SubstituteTypeMatchers`. Would this successfully match a `string[]` (runtime type) or a `IEnumerable<string>` against `IEnumerable<It.IsAnyType>`? Need to investigate.